doc/ui-guidelines.md

```diff
@@ -252,3 +265,150 @@ GOOD:
+
+By having an unstable reference, `Object.is` would always return `false` for reference values when comparing them between cycles. The implication of this is that when you pass a reference value with an unstable reference as a prop to a component that is memoized via `React.memo` (without a custom compare function), React would always see that prop as new value when comparing hence re-rendering the component because the reference to the prop changed even when the value hasn't essentially changed. The same is also true for `React.useMemo` and `React.useCallback`, if an unstable reference is passed as a dependency to them, the function they would recompute their function body and return a new reference to the memoized `value` or `callback`. By passing an unstable value (reference value) to these util functions, we essentially defeat the purpose of memoization in the first place.
+
+### Strategies for Creating Stable References
+
+How then do we create stable references? Well there are a number of ways, and there is no one size fits all way of doing it. But here are a few ways we could do it:
+#### Using global variables: 
+We can get stable references by declaring variables in a more global scope relative to the function's local scope. This essentially means if we can, we should declare the variables outside the functions scope. e.g:
+  ```clojure
+  (def some-map {:name "John Snow"
+                :knowledge 0})
+
+  (defn comp []
+    [rn/view {:some-map some-map}])
+  ```
+
+- Trade-offs: While global variables ensure reference stability, they can introduce side effects or global state management complexities. Overuse can make components harder to understand and test due to implicit dependencies.
+
+- Best Scenarios: Use for constants or configuration data that truly does not change over the application's lifetime and does not belong to any component's state.
+  
+#### Using `React.useRef`: 
+Provides a mutable ref object that remains constant throughout the component's lifecycle. Ideal for holding onto a value that does not trigger re-renders.. e.g
+  ```clojure
+  (defn comp []
+    (let [ref (rn/use-ref {:name "John Snow"
+                           :knowledge 0})]
+    [rn/view {:some-map (:current ref)}]))
+  ```
+- Trade-offs: It is not "reactive", meaning changes to its content do not cause the component to re-render. It's best used for values that are incidental to rendering.
```

Comment:
I don't believe this is a "trade-off", it's just the purpose it's used for.

Reply:
The "trade-off" meant here is just more of like a scenario where it doesn't make sense to use it

Reply:
perhaps something like "note" or "warning" would be better here then?